File: gcc/testsuite/gcc.dg/gomp/appendix-a/a.37.1.c
/* { dg-do compile } */

extern int omp_get_num_threads (void);
void work (int i);
void
incorrect ()
{
  int np, i;
  np = omp_get_num_threads ();	/* misplaced */
#pragma omp parallel for schedule(static)
  for (i = 0; i < np; i++)
    work (i);
}

File: gcc/testsuite/gfortran.dg/gomp/appendix-a/a.37.1.f90
! { dg-do compile }
      SUBROUTINE WORK(I)
      INTEGER I
        I=I+1
      END SUBROUTINE WORK
      SUBROUTINE INCORRECT()
        INTEGER OMP_GET_NUM_THREADS
        INTEGER I, NP
        NP = OMP_GET_NUM_THREADS()   !misplaced: will return 1
!$OMP PARALLEL DO SCHEDULE(STATIC)
          DO I = 0, NP-1
            CALL WORK(I)
          ENDDO
!$OMP END PARALLEL DO
      END SUBROUTINE INCORRECT
